### PR TITLE
feat(console): expose stack trace for console messages

### PR DESCRIPTION
**Pull request by hashseed**

### CODE REVIEW

File: src/common/ConsoleMessage.ts [hunk 1]
```diff
@@ -110,6 +110,13 @@ export class ConsoleMessage {
    * @returns The location of the console message.
    */
   location(): ConsoleMessageLocation {
-    return this._location;
+    return this._stackTraceLocations[0] ?? {};
```
Comment (mathiasbynens) [inline src/common/ConsoleMessage.ts]: We [support](https://github.com/puppeteer/puppeteer#usage) the latest maintenance Node.js LTS, so we cannot use nullish coalescing syntax in the compiled JS yet. Let's double-check that the generated output works in Node.js v10. cc @jackfranklin

**Review by mathiasbynens — APPROVED** [review]

File: src/common/ConsoleMessage.ts (hunk 1 above)
Comment (jackfranklin) [reply src/common/ConsoleMessage.ts]: If CI passes we should be good as that compiles and runs on Node 10. I think our ts-config is set up to compile this down correctly but we will see!

**Review by jackfranklin — COMMENTED** [review]
I think you'll also need to do `npm run generate-docs` to regenerate the TSDoc output (which is the new docs we are going to migrate to in time).

File: src/common/ConsoleMessage.ts (hunk 1 above)
Comment (hashseed) [reply src/common/ConsoleMessage.ts]: the build artifacts in `lib/` still has `??`. Let me rephrase this in outdated JS.

Comment (jackfranklin) [reply src/common/ConsoleMessage.ts]: Looks like TypeScript is not compiling it away. I can look into fixing it or I'd rewrite to use a ternary (or similar) to avoid this for now, to be honest.

**Comment by hashseed** [comment]
I ran `npm run generate-docs` and replaced `??` with a ternary!

File: src/common/ConsoleMessage.ts (hunk 1 above)
Comment (mathiasbynens) [reply src/common/ConsoleMessage.ts]: Longer-term it might make sense to fix our tsconfig to align with the maintenance Node.js LTS version, so we don't have to worry about this in the future.

**Review by mathiasbynens — APPROVED** [review]
